nubia/internal/nubia.py

```diff
@@ -309,7 +309,7 @@ def run(self, cli_args=sys.argv, ipython=False):
             return self.start_ipython(args)
         # by default, if no command is passed we will get 'connect'
         if args._cmd == "connect":
-            return self.start_interactive(args)
+            return await self.start_interactive(args)
         else:
             ret = self.run_cli(args)
```

Comment:
You would have to make the non-interactive side async as well. This would lead to line 271 where you have to make it async.

Reply:
Are you sure ? All `run_cli` methods (which you point to when you mention line 271) are sync methods. [I have not changed the way they work in the `AutoCommand` class](https://github.com/esciara/python-nubia/blob/9a5d03b7fa050dbe46c8893ef890ec1d6a8adeb8/nubia/internal/cmdbase.py#L459). 

The [command `"connect"` specifically represents the interactive mode](https://github.com/facebookincubator/python-nubia/blob/eae1b64338ee08dbf2a81ec8891777b039ba9085/nubia/internal/commands/builtin.py#L21). Hence, non-interactive would not be concerned by this change. I actually left alone the non-interactive mode alone on purpose, and work on not changing anything that would affect it. So if anything had to break, it would be in the interactive mode, and I have changed the tests to ensure this is not the case.

The only breaking change is that [the `shell.run()` method has to be called in a loop](https://github.com/esciara/python-nubia/blob/9a5d03b7fa050dbe46c8893ef890ec1d6a8adeb8/example/nubia_example.py#L29).

Reply:
I think that if we want to add async support to Nubia, that we need to be able to be able to have all or most of the entrypoint methods be async (and support sync for backwards compatibility). It would be a tad strange if `run_interactive` allowed for `async` methods, but `run_cli` did not.